compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -612,3 +613,46 @@ OMR::ARM64::TreeEvaluator::lxorEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	// boolean xor of 2 integers
 	return genericBinaryEvaluator(node, TR::InstOpCode::eorx, TR::InstOpCode::eorimmx, true, cg);
 	}
+
+// Also handles TR::aluadd
+TR::Register *
+OMR::ARM64::TreeEvaluator::aladdEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register *trgReg = laddEvaluator(node, cg);
+
+   if (node->isInternalPointer())
+      {
+      if (node->getPinningArrayPointer())
+         {
+         trgReg->setContainsInternalPointer();
+         trgReg->setPinningArrayPointer(node->getPinningArrayPointer());
+         }
+      else
+         {
+         TR::Node *firstChild = node->getFirstChild();
+         if ((firstChild->getOpCodeValue() == TR::aload) &&
+             firstChild->getSymbolReference()->getSymbol()->isAuto() &&
+             firstChild->getSymbolReference()->getSymbol()->isPinningArrayPointer())
+            {
+            trgReg->setContainsInternalPointer();
+
+            if (!firstChild->getSymbolReference()->getSymbol()->isInternalPointer())
+               {
+               trgReg->setPinningArrayPointer(firstChild->getSymbolReference()->getSymbol()->castToAutoSymbol());
+               }
+            else
+               {
+               trgReg->setPinningArrayPointer(firstChild->getSymbolReference()->getSymbol()->castToInternalPointerAutoSymbol()->getPinningArrayPointer());
+               }
+            }
+         else if (firstChild->getRegister() &&
+                  firstChild->getRegister()->containsInternalPointer())
+            {
+            trgReg->setContainsInternalPointer();
+            trgReg->setPinningArrayPointer(firstChild->getRegister()->getPinningArrayPointer());
+            }
+         }
+      }
```

Comment:
I think this will accomplish the same thing for the code under the `isInternalPointer()` guard but with fewer calls and indirections.  Correct me if I'm wrong though.  This wouldn't work, for example, if the pinning array pointer symbol could ever be NULL, but I don't think that's possible.
```
   if (node->isInternalPointer())
      {
      TR::AutomaticSymbol *pinningArrayPointerSymbol = NULL;

      if (node->getPinningArrayPointer())
         {
         pinningArrayPointerSymbol = node->getPinningArrayPointer();
         }
      else
         {
         TR::Node *firstChild = node->getFirstChild();
   
         if ((firstChild->getOpCodeValue() == TR::aload) &&
             firstChild->getSymbolReference()->getSymbol()->isAuto() &&
             firstChild->getSymbolReference()->getSymbol()->isPinningArrayPointer())
            {
            TR::Symbol *firstChildSymbol = firstChild->getSymbolReference()->getSymbol();
            
            pinningArrayPointerSymbol = firstChildSymbol->isInternalPointer() ?
               firstChildSymbol->castToInternalPointerAutoSymbol()->getPinningArrayPointer() :
               firstChildSymbol->castToAutoSymbol();
            }
         else if (firstChild->getRegister() &&
                  firstChild->getRegister()->containsInternalPointer())
            {
            pinningArrayPointerSymbol = firstChild->getRegister()->getPinningArrayPointer();
            }
         }

      if (pinningArrayPointerSymbol)
         {
         trgReg->setContainsInternalPointer();
         trgReg->setPinningArrayPointer(pinningArrayPointerSymbol);
         }
      }
```